src/meds_etl/simple_csv.py

```diff
@@ -0,0 +1,220 @@
+            )
+            numeric_value = value.cast(pl.Float32, strict=False)
+
+            text_value = (
+                pl.when(datetime_value.is_null() & numeric_value.is_null() & (value != pl.lit("")))
+                .then(value)
+                .otherwise(pl.lit(None, dtype=str))
+            )
+
+            metadata = {}
+
+            for colname in batch.columns:
+                if colname in ("patient_id", "time", "code", "value"):
+                    continue
+                metadata[colname] = pl.col(colname)
+
+            def transform_metadata(d):
+                return pl.struct([v.alias(k) for k, v in d.items()])
+
+            metadata = transform_metadata(metadata)
+
+            batch = batch.filter(code.is_not_null())
+
+            event_data = (
+                batch.select(
+                    patient_id=patient_id,
+                    time=time,
+                    code=code,
+                    text_value=text_value,
+                    numeric_value=numeric_value,
+                    datetime_value=datetime_value,
+                    metadata=metadata,
+                    shard=patient_id.hash(213345) % num_shards,
+                )
+                .collect()
+                .partition_by("shard", as_dict=True, maintain_order=False)
+            )
+
+            for shard_index, shard in event_data.items():
+                fname = os.path.join(
+                    temp_dir, str(shard_index), f'{index}_{batch_index}.parquet'
+                )
+                shard.write_parquet(fname, compression="uncompressed")
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="meds_etl_simple_csv", description="Performs an ETL from a simple flat CSV format to MEDS")
+    parser.add_argument("src_csv", type=str)
+    parser.add_argument("destination", type=str)
+    parser.add_argument("--num_shards", type=int, default=100)
+    parser.add_argument("--num_threads", type=int, default=1)
+    args = parser.parse_args()
+
+    if not os.path.exists(args.src_csv):
+        raise ValueError(f'The source CSV folder ("{args.src_csv}") does not seem to exist?')
+
+    os.makedirs(args.destination, exist_ok=True)
+
+    events = []
+
+    decompressed_dir = os.path.join(args.destination, "decompressed")
+    os.mkdir(decompressed_dir)
+
+    temp_dir = os.path.join(args.destination, "temp")
+    os.mkdir(temp_dir)
+
+    for shard_index in range(args.num_shards):
+        os.mkdir(os.path.join(temp_dir, str(shard_index)))
+
+    metadata = {
+        "dataset_name": args.src_csv,
+        "dataset_version": "unknown",
+        "etl_name": "meds_etl.simple_csv",
+        "etl_version": meds_etl.__version__,
+    }
+
+    jsonschema.validate(instance=metadata, schema=meds.dataset_metadata)
+
+    with open(os.path.join(args.destination, "metadata.json"), "w") as f:
+        json.dump(metadata, f)
+
+    all_tasks = []
+
+    random.seed(3422342)
+    random.shuffle(all_tasks)
```

Comment:
This is probably meant to be in a different line, right?

Reply:
Oops. Fixed.